### DISCUSSION TITLE
Question about SpecifiedSmoothCircleIC

### DISCUSSION THREAD

**Discussion by abc-hy** [idaholab/moose#24568] [Q&A Modules: Phase field] [answered]
Hi everyone,

I have a question about using SpecifiedSmoothCircleIC to set the initial condition of circle. I want to generate a initial condition like this,

![image](https://github.com/idaholab/moose/assets/63679626/66b86145-5627-436c-a22c-cca6d813c893)
In the most part of the square model, the initial c is set as 0, however, in some small circles, the initial condition should be set as larger values. How can I randomly generate some values for the circle's initial concentration in some range? I saw that in all the examples shown in SpecifiedSmoothCircleIC webpage, the initial concentration can only be set as one specific value. How can I set different values for each circle's concentration? Or I can only set the same value for the two circles?

![image](https://github.com/idaholab/moose/assets/63679626/5e4f03c7-dd24-4942-82b9-cc69d95507c6)

Thank you,
Best,

**Comment by ykvishal** [comment, marked as answer]
You could try with CoupledValueFunctionIC, RandomIC, SpecifiedSmoothCircleIC. Use 1 temp initial variable to populate random values with RandomIC. Then use another temp variable to create circles shown in Fig with SpecifiedSmoothCircleIC; use 1 and 0 values for this case. Finally, user CoupledValueFunctionIC for main variable and multiply two temp variables. I hope that it may work.
